libpkg/pkg_add.c

```diff
@@ -186,14 +187,18 @@ get_uid_from_archive(struct archive_entry *ae)
 	const char *user;
 	static struct passwd pwent;
 	struct passwd *result;
+	int err;
 
 	user = archive_entry_uname(ae);
 	if (pwent.pw_name != NULL && strcmp(user, pwent.pw_name) == 0)
 		goto out;
 	pwent.pw_name = NULL;
-	if ((getpwnam_r(user, &pwent, user_buffer, sizeof(user_buffer),
-	    &result)) < 0)
+	err = getpwnam_r(user, &pwent, user_buffer, sizeof(user_buffer),
+	    &result);
+	if (err != 0) {
+		warnc(err, "getpwnam_r");
```

Comment:
you should use pkg_emit* and not warn* in the library, I will fix it after merge